modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js

```diff
@@ -111,6 +111,60 @@ export function handleAction(
 
 		window.open(url);
 	}
+	else if (target === 'download') {
+		event.preventDefault();
+
+		setLoading(true);
+		let filename = '';
+
+		executeAsyncAction(url, method)
+			.then((response) => {
+				const disposition = response.headers.get('Content-Disposition');
+
+				if (disposition && disposition.indexOf('attachment') !== -1) {
+					const filenameRegex = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/;
+					const matchesRegex = filenameRegex.exec(disposition);
+					if (matchesRegex !== null && matchesRegex[1]) {
+						filename = matchesRegex[1].replace(/['"]/g, '');
+					}
+				}
+
+				return response.blob();
+			})
+			.then((responseBlob) => {
+				const downloadUrl = URL.createObjectURL(responseBlob);
+				const downloadElement = document.createElement('a');
+
+				if (
+					filename &&
+					typeof downloadElement.download !== 'undefined'
+				) {
+					downloadElement.href = downloadUrl;
+					downloadElement.download = filename;
+					document.body.appendChild(downloadElement);
+					downloadElement.click();
+				}
+				else {
+					window.location.href = downloadUrl;
+				}
+
+				openToast({
+					message:
+						successMessage ||
+						Liferay.Language.get('action-completed'),
+					type: 'success',
+				});
+
+				setLoading(false);
+			})
+			.catch((_) => {
```

Comment:
If this is not being used can we remove the `_`?

Reply:
I'm not sure, I borrowed this from the async case which did the same thing.

Reply:
I'd remove it since it doesn't serve any purpose imo. It'd make more sense if you were accessing a second attribute like `(_, message)` and you weren't using the first value and it needed to be defined.